### PR TITLE
chore: enable `no-unsafe-call` locally

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
Continues #3278.

`isomorphic-fetch`'s types rely on a global `fetch` to already exist. `node-fetch` is already a dev dependency so I switched the `generate-contributors` calls to it.

### CODE REVIEW

File: packages/experimental-utils/src/eslint-utils/RuleTester.ts [hunk 1]
```diff
@@ -32,6 +32,7 @@ class RuleTester extends TSESLint.RuleTester {
       try {
         // instead of creating a hard dependency, just use a soft require
         // a bit weird, but if they're using this tooling, it'll be installed
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call
         require(parser).clearCaches();
```
Comment (bradzacher) [inline packages/experimental-utils/src/eslint-utils/RuleTester.ts:36]: Do we want to add an explicit type assertion here to ensure we statically track this usage?

**Review by bradzacher — APPROVED** [review]

File: packages/experimental-utils/src/eslint-utils/RuleTester.ts (hunk 1 above)
Comment (JoshuaKGoldberg) [reply packages/experimental-utils/src/eslint-utils/RuleTester.ts:36]: Mmhh I like that!

Comment (JoshuaKGoldberg) [reply packages/experimental-utils/src/eslint-utils/RuleTester.ts:36]: Aww, you can't do this even with a known literal type for `parser`:

```ts
        (require(parser) as typeof import(typeof parser)).clearCaches();
```

I'm.. not going to bother filing an issue on TypeScript for that one.

Comment (JoshuaKGoldberg) [reply packages/experimental-utils/src/eslint-utils/RuleTester.ts]: Hmm, [build failures](https://github.com/typescript-eslint/typescript-eslint/pull/3281/checks?check_run_id=2348457824):

```
Cannot find module '@typescript-eslint/parser' or its corresponding type declarations.
```

`parser` has a tsconfig build dependency on `experimental-utils` so adding one is a no-go. `experimental-utils` _does_ already have a dependency on `typescript-estree` though, so... this works!

```
        (require(parser) as typeof import('@typescript-eslint/typescript-estree')).clearCaches();
```

Comment (JoshuaKGoldberg) [reply packages/experimental-utils/src/eslint-utils/RuleTester.ts]: Looking back, having this be the `typeof` a _different_ module from what it's actually importing is kinda weird. I'll revert and leave this thread open for you to decide...

Comment (bradzacher) [reply packages/experimental-utils/src/eslint-utils/RuleTester.ts:36]: I'm happy to leave this untyped if it's too shitty to make it work - no problem!

**Review by bradzacher — APPROVED** [review]

**Comment by bradzacher** [comment]
Thanks! 🥇

**Pull request merged**